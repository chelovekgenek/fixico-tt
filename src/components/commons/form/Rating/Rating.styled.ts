import styled from "styled-components";

import { Box, Typography } from "@material-ui/core";

export const BoxWrapper = styled(Box)`
  padding: 0;
  margin: 0;
  border: none;
`;

export const TypographyWrapper = styled(Typography)`
  padding: 0;
`;
