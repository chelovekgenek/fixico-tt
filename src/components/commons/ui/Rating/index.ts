export { Rating } from "@material-ui/lab";
