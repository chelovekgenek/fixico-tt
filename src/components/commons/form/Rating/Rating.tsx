import React from "react";
import { Field, FieldProps } from "formik";

import { Rating as MUIRating } from "components/commons/ui";

import { BoxWrapper, TypographyWrapper } from "./Rating.styled";

interface IProps {
  name: string;
  label: string;
}

export const Rating: React.FC<IProps> = ({ name, label }) => (
  <Field name={name}>
    {(props: FieldProps) => (
      <InnerFormItem name={name} label={label} {...props} />
    )}
  </Field>
);

export const InnerFormItem: React.FC<IProps & FieldProps> = ({
  label,
  name,
  field,
  form,
  ...props
}) => (
  <BoxWrapper component="fieldset" borderColor="transparent">
    <TypographyWrapper component="legend">{label}</TypographyWrapper>
    <MUIRating
      id={name}
      name={name}
      value={field.value}
      onChange={field.onChange}
      onBlur={field.onBlur}
    />
  </BoxWrapper>
);
