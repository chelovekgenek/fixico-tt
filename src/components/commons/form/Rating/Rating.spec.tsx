import React from "react";
import { shallow } from "enzyme";

import fieldProps from "mocks/formik-field-props";

import { InnerFormItem } from "./Rating";

describe("Rating", () => {
  const defaultProps = {
    ...fieldProps,
    field: { ...fieldProps.field, value: 5, name: "rating" },
    name: "rating",
    label: "rating",
    value: 5
  };
  const render = (
    props: Partial<React.ComponentProps<typeof InnerFormItem>> = {}
  ) => shallow(<InnerFormItem {...defaultProps} {...props} />);

  it("should render", () => {
    expect(render()).toMatchSnapshot();
  });
});
