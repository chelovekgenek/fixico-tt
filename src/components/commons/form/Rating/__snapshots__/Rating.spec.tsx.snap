// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render 1`] = `
<Styled(Styled(MuiBox))
  borderColor="transparent"
  component="fieldset"
>
  <Styled(WithStyles(ForwardRef(Typography)))
    component="legend"
  >
    rating
  </Styled(WithStyles(ForwardRef(Typography)))>
  <WithStyles(ForwardRef(Rating))
    id="rating"
    name="rating"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    value={5}
  />
</Styled(Styled(MuiBox))>
`;
